src/validation/regex/regex.validation.ts

```diff
@@ -2,3 +2,5 @@ export const VALID_CHARACTERS = /^[-,.:; 0-9A-Z&@$£¥€'"«»''""?!/\\()[\]{}<
 
 // same as VALID_CHARACTERS but also allows carriage returns
 export const VALID_CHARACTERS_FOR_TEXT_BOX = /^[-,.:; 0-9A-Z&@$£¥€'"«»''""?!/\\()[\]{}<>*=#%+ÀÁÂÃÄÅĀĂĄÆǼÇĆĈĊČÞĎÐÈÉÊËĒĔĖĘĚĜĞĠĢĤĦÌÍÎÏĨĪĬĮİĴĶĹĻĽĿŁÑŃŅŇŊÒÓÔÕÖØŌŎŐǾŒŔŖŘŚŜŞŠŢŤŦÙÚÛÜŨŪŬŮŰŲŴẀẂẄỲÝŶŸŹŻŽa-zÀÖØſƒǺẀỲàáâãäåāăąæǽçćĉċčþďðèéêëēĕėęěĝģğġĥħìíîïĩīĭįĵķĺļľŀłñńņňŋòóôõöøōŏőǿœŕŗřśŝşšţťŧùúûüũūŭůűųŵẁẃẅỳýŷÿźżž\r\n]*$/;
+
+export const VALID_EMAIL_FORMAT = /^[a-zA-Z0-9_+&*-]+(?:\.[a-zA-Z0-9_+&*-]+)*@(?:[a-zA-Z0-9-]+\.)+[a-zA-Z]{2,}$/;
```

Comment:
the regex here doesn't match the one in the API when compared side by side
^[a-zA-Z0-9_+&*-]+(?:\.[a-zA-Z0-9_+&*-]+)*+@(?:[a-zA-Z0-9-]+\.)++[a-zA-Z]{2,15}$
^[a-zA-Z0-9_+&*-]+(?:\.[a-zA-Z0-9_+&*-]+)*@(?:[a-zA-Z0-9-]+\.)+[a-zA-Z]{2,}$

Reply:
we should make the web regex the same as the on from the API

Reply:
It matches https://owasp.org/www-community/OWASP_Validation_Regex_Repository, the one it the api is the exception. 

^[a-zA-Z0-9_+&-]+(?:.[a-zA-Z0-9_+&-]+)+@(?:[a-zA-Z0-9-]+.)++[a-zA-Z]{2,15}$
does not work in the web or in the online regex testers with the examples on the ticket. Would you be ok to take a look and verify?

Reply:
The API regex seems to work ok for me, it matches the 6 examples in the ticket 
I'll attach a screenshot
The api regex has double slashes instead of one to escape the slash inside the string
so it should look like this without the escaping
`^[a-zA-Z0-9_+&*-]+(?:\.[a-zA-Z0-9_+&*-]+)*+@(?:[a-zA-Z0-9-]+\.)++[a-zA-Z]{2,15}$`
The comments in the API regex say that we had to modify the OWASP regex due to Sonar highlighting inefficiencies - https://github.com/companieshouse/overseas-entities-api/blob/main/src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/utils/StringValidators.java#L50
<img width="1783" alt="Screenshot 2022-11-23 at 15 16 12" src="https://user-images.githubusercontent.com/11330756/203584368-b4552459-e331-48aa-aefc-fc6f476cdbe2.png">

Reply:
i'll get your branch and have a look, hopefully it is just an escaping issue

Reply:
Just seen that myself in the api, but this creates complications for the web, there may have to be a change as to what we are aiming for here. Not sure what we want to do.

Reply:
Hold I'll take a look without the escaping.

Reply:
Ok, this is what I used:
/^[a-zA-Z0-9_+&*-]+(?:\.[a-zA-Z0-9_+&*-]+)*@(?:[a-zA-Z0-9-]+\.)+[a-zA-Z]{2,15}$/
and the tests pass, but I did have to removed two extra '+' characters  one just before the '@' not sure what they are for in the api.

Reply:
Just seen the note in the api about the '+'s, but the web really does not like them.

Reply:
yeah, just experienced the same issue , it seems the compiler doesn't like our regex
`SyntaxError: Invalid regular expression: /^[a-zA-Z0-9_+&*-]+(?:\.[a-zA-Z0-9_+&*-]+)*+@(?:[a-zA-Z0-9-]+\.)++[a-zA-Z]{2,15}$/: Nothing to repeat`
hmm, i'll do some googling but we'll have to have the web and api regexes matching, might end up updating the api one too.

Reply:
Sonar seems happy.

Reply:
@markpit can this (long) conversation now be resolved/collapsed?